Package fails to install due to pylibmc

In [this commit](https://github.com/pytest-dev/pytest-services/commit/2c02eb56bfc4c8b3af41d82536535b1b2ada83ee) a dependency on pylibmc was added. When I try to build the package on my Ubuntu Trusty box, the package fails to install thus:

```
vagrant@trusty:~$ pip install -t out pytest-services-1.1.14.tar.gz 
Unpacking ./pytest-services-1.1.14.tar.gz
  Running setup.py (path:/tmp/pip-jqgBF9-build/setup.py) egg_info for package from file:///home/vagrant/pytest-services-1.1.14.tar.gz
    
Downloading/unpacking setuptools (from pytest-services==1.1.14)
  Downloading setuptools-35.0.2-py2.py3-none-any.whl (390kB): 390kB downloaded
Downloading/unpacking requests (from pytest-services==1.1.14)
  Downloading requests-2.14.2-py2.py3-none-any.whl (560kB): 560kB downloaded
Downloading/unpacking psutil (from pytest-services==1.1.14)
  Downloading psutil-5.2.2.tar.gz (348kB): 348kB downloaded
  Running setup.py (path:/tmp/pip_build_vagrant/psutil/setup.py) egg_info for package psutil
    
    warning: manifest_maker: MANIFEST.in, line 14: 'recursive-include' expects <dir> <pattern1> <pattern2> ...
    
    warning: no previously-included files matching '*' found under directory 'docs/_build'
    warning: no previously-included files matching '*' found under directory '.ci'
Downloading/unpacking pylibmc (from pytest-services==1.1.14)
  Downloading pylibmc-1.5.2.tar.gz (62kB): 62kB downloaded
  Running setup.py (path:/tmp/pip_build_vagrant/pylibmc/setup.py) egg_info for package pylibmc
    
    warning: no files found matching 'runtests.py'
    warning: no files found matching '*.py' under directory 'pylibmc'
Downloading/unpacking pytest (from pytest-services==1.1.14)
  Downloading pytest-3.0.7-py2.py3-none-any.whl (172kB): 172kB downloaded
Downloading/unpacking subprocess32 (from pytest-services==1.1.14)
  Downloading subprocess32-3.2.7.tar.gz (54kB): 54kB downloaded
  Running setup.py (path:/tmp/pip_build_vagrant/subprocess32/setup.py) egg_info for package subprocess32
    
    warning: no files found matching '*.h'
    no previously-included directories found matching 'build'
    no previously-included directories found matching 'dist'
    no previously-included directories found matching '.hg*'
Downloading/unpacking appdirs>=1.4.0 (from setuptools->pytest-services==1.1.14)
  Downloading appdirs-1.4.3-py2.py3-none-any.whl
Downloading/unpacking six>=1.6.0 (from setuptools->pytest-services==1.1.14)
  Downloading six-1.10.0-py2.py3-none-any.whl
Downloading/unpacking packaging>=16.8 (from setuptools->pytest-services==1.1.14)
  Downloading packaging-16.8-py2.py3-none-any.whl
Downloading/unpacking py>=1.4.29 (from pytest->pytest-services==1.1.14)
  Downloading py-1.4.33-py2.py3-none-any.whl (83kB): 83kB downloaded
Downloading/unpacking pyparsing (from packaging>=16.8->setuptools->pytest-services==1.1.14)
  Downloading pyparsing-2.2.0-py2.py3-none-any.whl (56kB): 56kB downloaded
Installing collected packages: setuptools, requests, psutil, pylibmc, pytest, subprocess32, pytest-services, appdirs, six, packaging, py, pyparsing
  Running setup.py install for psutil
    building 'psutil._psutil_linux' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_VERSION=522 -DPSUTIL_LINUX=1 -I/usr/include/python2.7 -c psutil/_psutil_linux.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o
    x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o -o build/lib.linux-x86_64-2.7/psutil/_psutil_linux.so
    building 'psutil._psutil_posix' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_VERSION=522 -DPSUTIL_LINUX=1 -I/usr/include/python2.7 -c psutil/_psutil_posix.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_posix.o
    x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/psutil/_psutil_posix.o -o build/lib.linux-x86_64-2.7/psutil/_psutil_posix.so
    
    warning: manifest_maker: MANIFEST.in, line 14: 'recursive-include' expects <dir> <pattern1> <pattern2> ...
    
    warning: no previously-included files matching '*' found under directory 'docs/_build'
    warning: no previously-included files matching '*' found under directory '.ci'
  Running setup.py install for pylibmc
    building '_pylibmc' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE_ZLIB -I/usr/include/python2.7 -c src/_pylibmcmodule.c -o build/temp.linux-x86_64-2.7/src/_pylibmcmodule.o -fno-strict-aliasing -std=c99
    In file included from src/_pylibmcmodule.c:34:0:
    src/_pylibmcmodule.h:42:36: fatal error: libmemcached/memcached.h: No such file or directory
     #include <libmemcached/memcached.h>
                                        ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_vagrant/pylibmc/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-hpGDdF-record/install-record.txt --single-version-externally-managed --compile --home=/tmp/tmpfLJkGn:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/__main__.py -> build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/pools.py -> build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/test.py -> build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/consts.py -> build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/client.py -> build/lib.linux-x86_64-2.7/pylibmc

copying src/pylibmc/__init__.py -> build/lib.linux-x86_64-2.7/pylibmc

running build_ext

building '_pylibmc' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/src

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE_ZLIB -I/usr/include/python2.7 -c src/_pylibmcmodule.c -o build/temp.linux-x86_64-2.7/src/_pylibmcmodule.o -fno-strict-aliasing -std=c99

In file included from src/_pylibmcmodule.c:34:0:

src/_pylibmcmodule.h:42:36: fatal error: libmemcached/memcached.h: No such file or directory

 #include <libmemcached/memcached.h>

                                    ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_vagrant/pylibmc/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-hpGDdF-record/install-record.txt --single-version-externally-managed --compile --home=/tmp/tmpfLJkGn failed with error code 1 in /tmp/pip_build_vagrant/pylibmc
Storing debug log for failure in /home/vagrant/.pip/pip.log
```

pylibmc shouldn't be a dependency, especially if it adds a compiled dependency and especially if that library has compile-time dependencies.

--- a/README.rst
+++ b/README.rst
@@ -122,7 +122,11 @@ Service fixtures
 * memcached_connection
     Memcached connection string.
 * do_memcached_clean
-    Determine if memcached should be cleared before every test run. Equals to `run_services` fixture by default.
+    Determine if memcached should be cleared before every test run. Equals to `run_services` fixture by default. Requires `pylibmc` installed or `memcache` indicated
+    as an extra.
+* memcached_client
+    A pylibmc.Client instance bound to the service. Requires `pylibmc` installed
+    or `memcache` indicated as an extra.
 * mysql
     Start mysql-server_ instance.
 * mysql_database_name

--- a/pytest_services/memcached.py
+++ b/pytest_services/memcached.py
@@ -2,8 +2,6 @@
 import os
 import pytest
 
-from pylibmc import Client
-
 
 @pytest.fixture(scope='session')
 def memcached_socket(run_dir, run_services):
@@ -38,7 +36,8 @@ def do_memcached_clean(run_services):
 @pytest.fixture(scope='session')
 def memcached_client(memcached_socket, memcached):
     """Create client for memcached."""
-    return Client([memcached_socket])
+    mc = pytest.importorskip('pylibmc')
+    return mc.Client([memcached_socket])
 
 
 @pytest.fixture

--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,6 @@ install_requires = [
     'setuptools',
     'requests',
     'psutil',
-    'pylibmc',
     'pytest',
 ]
 
@@ -38,6 +37,9 @@ setup(
     version=pytest_services.__version__,
     url='https://github.com/pytest-dev/pytest-services',
     install_requires=install_requires,
+    extras={
+        'memcached': ['pylibmc'],
+    },
     classifiers=[
         'Development Status :: 6 - Mature',
         'Intended Audience :: Developers',

--- a/tox.ini
+++ b/tox.ini
@@ -9,6 +9,7 @@ deps =
     -e.
     -r{toxinidir}/requirements-testing.txt
 passenv = DISPLAY COVERALLS_REPO_TOKEN USER TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
+extras = memcache
 
 [testenv:linters]
 basepython=python2.7

